scripts/kind-e2e/lib_subctl_gather_test.sh

```diff
@@ -89,18 +92,27 @@ function validate_resource_files() {
   local kind=$3
   local selector=$4
   local cluster_name=$5
+  local nsarg="--namespace=${ns}"
+
+  if [[ "$ns" == "all" ]]; then
+    nsarg="-A"
+  fi
 
   if [[ $cluster_name == "" ]]; then
     cluster_name=${cluster}
   fi
 
-  names=$(kubectl get $resource --namespace=$ns $selector -o=jsonpath='{.items..metadata.name}')
+  names=$(kubectl get $resource $nsarg $selector -o=jsonpath='{.items..metadata.name}')
+  namespaces=$(kubectl get $resource $nsarg $selector -o=jsonpath='{.items..metadata.namespace}')
```

Comment:
This has potential to run into problems where number of entries don't match coz some got deleted between these two `get` commands. e.g. Gather test is run at end of e2e when namespace is deleted and objects are being deleted but not all gone yet. Not sure if we can perfectly address the issue, but would be good to echo some warning message if the no. of items in these two don't match to help troubleshoot any flaky failures.

Reply:
I changed it to capture the whole JSON and use `jq` to parse the fields.